rfc/003-remote-pipelines/README.md

```diff
@@ -0,0 +1,206 @@
+# Remote Pipelines
+
+- **Status**: WIP
+- **Created**: Sep 14, 2022
+- **ETA**: Oct 14, 2022
+- **Authors**:
+  - [Matthias Vallentin](https://github.com/mavam)
+- **Discussion**: [PR #2577](https://github.com/tenzir/vast/pull/2577)
+
+## Overview
+
+This proposal enhances the local pipeline execution with a remote component,
+making it possible to connect pipeline across process boundaries.
+
+## Problem Statement
+
+As of RFC-001, VAST pipelines can execute locally. But a one-shot pipeline
+execution is not sufficient. After this building block, natural questions arise:
+
+- How can I connect a local and remote pipeline?
+- How can I manage (add/update/remove) a set of continuously running pipelines?
+- How do pipelines interact with VAST server nodes? Specifically, how can I send
+  data into a VAST node, and how can I query data from a VAST node?
+
+This proposal seeks an answer to these questions.
+
+## Solution Proposal
+
+The core of this proposal introduces a control plane that provides a global
+namespace for pipelines and VAST nodes. The ability to interconnect pipelines
+via unique names in location-independent manner enabeles a loosely coupled mesh
+arcitecture.
+
+The core primitives for establishing connections between pipelines are two
+dedicated operators, `push` and `pull`:
+
+| Operator | Input Type | Output Type | Description
+| -------- | ---------- | ----------- | ------------------------------------------
+| `push`   | `Arrow`    | `Void`      | Sends data to a named pipeline
+| `pull`   | `Void`     | `Arrow`     | Attaches to a named pipeline
+
+Arguments to these operators represent names from the global namespace.
+
+### Example
+
+Assuming an existing mechanism to bind a pipeline to name, the new operators
+allow for interconnecting pipelines as follows.
+
+Create a named pipeline `pipe1` at a VAST node and expose in the global control
+plane:
+
+```bash
+vast spawn pipe1 'from - | read zeek | where orig_bytes > 1 GiB'
+vast expose pipe1 tcp://1.2.3.4:5555 
```

Comment:
Is this a leftover from a previous design? I thought from RFC-001 that `from`/`read` where now relegated to the realm of the unexposed physical operators.

Also I am not sure what `vast expose` does here. Is it attaching to the input or the ouput? 

Finally, I believe that the address (`tcp://...`) should not be an argument to the `expose` command but rather something that provided in the response